threads.go

```diff
@@ -737,7 +758,7 @@ func (t *threads) getLocalRecords(
 		if !cursor.Defined() || cursor.String() == offset.String() {
 			break
 		}
-		r, err := cbor.GetRecord(ctx, t, cursor, fk)
+		r, err := cbor.GetRecord(ctx, t, cursor, fk) // Important invariant: heads are always in blockstore
```

Comment:
Boom!
+1 for the adding a height / sequence number to the records. Dealing with multiple blockstores could become cumbersome. Also, I think a single blockstore will be nice if we ever integrate with ipfs-cluster.

Reply:
Cool, created #137

Reply:
Replied too fast.
We should try to do this in the short-term, since the current situation is quite brittle. Any change we make that might fetch records and save it to the blockstore may confuse the `putRecord`.

Reply:
Cool. +1